pkg/metrics/series.go

```diff
@@ -48,7 +48,7 @@ func (p *Point) MarshalJSON() ([]byte, error) {
 type Serie struct {
 	Name           string          `json:"metric"`
 	Points         []Point         `json:"points"`
-	Tags           []string        `json:"tags"`
+	Tags           *CompositeTags  `json:"tags"`
```

Comment:
When I wrote this code, I was wondering if I should use `*CompositeTag` or `CompositeTag`. I choose pointer. After a second though, I agree with you `CompositeTag` is more appropriate as by design `CompositeTag` is small and immutable.

Reply:
`CompositeTags` implements `UnmarshalJSON` which requires a pointer to be used (otherwise the instance is never updated). 
Replacing `*CompositeTags` by `CompositeTags` means a `Serie` cannot be unmarshalled.
I have noticed `MarshalJSON()` is implemented for `Series` but there is no `UnmarshalJSON` which means it may be OK if `Serie` cannot be unmarshalled. The problem is that there is a no easy to be sure `UnmarshalJSON` is not called somewhere in very specific code path.

Reply:
> CompositeTags implements UnmarshalJSON which requires a pointer to be used (otherwise the instance is never updated).
> Replacing *CompositeTags by CompositeTags means a Serie cannot be unmarshalled.

Not at all. The fact that struct is embedded does not prevent you from calling methods with a pointer receiver; the receiver will point to the parent struct's field: https://go.dev/play/p/Hf-xF0r-drX

Reply:
Good to know!